joomla/joomla-cms
# [4.0] Remove Exception-catching from JMail

### Summary of Changes
Remove the internal handling of exceptions thrown by the PHPMailer API in JMail based on PR #12179. 

All the proposed changes in that PR were implemented, furthermore we added Exception Handling in the places that use the methods of the JMail class.

### Testing Instructions
Mail sending should still work as intended, and with exceptions turned off all booleans should be handled normally as well.

With exceptions turned on all mail functions should now show appropriate warning messages and/or execute expected error behaviour.

Functionalities updated:
- Notification Email for a direct message
- Admin Mass-Mail
- Joomla! update notification mail (could not be tested)
- Sending a mail through the Contact Form
- Registration Email after a user is created in the backend
- Mailto function of articles (this will not show an error anymore, but will just remain on the sending form, might require more work)
- Remind/Reset emails for the username and password

### Documentation Changes Required
Error handling and throwing behaviour changed.
